File: Practica2_i-Lunch/src/gestorAplicacion/usuariosRestaurante/cliente.py
from gestorAplicacion.usuariosRestaurante.usuario import Usuario

class Cliente(Usuario):

    # Atributos estaticos

    _clientes = []

    # Constructor

    def __init__(self, nombre = "", email = "", telefono = 0, direccion = "", edad = 0):
        self._nombre = nombre
        self._email = email
        self._telefono = telefono
        self._direccion = direccion
        self._edad = edad

        self._historialPedidos = []

        Cliente._clientes.append(self)

    # Getters y Setters

    def getNombre(self):
        return self._nombre

    def setNombre(self, nombre):
        self._nombre = nombre

    def getEmail(self):
        return self._email

    def setEmail(self, email):
        self._email = email
    
    def getTelefono(self):
        return self._telefono

    def setTelefono(self, telefono):
        self._telefono = telefono

    def getDireccion(self):
        return self._direccion

    def setDireccion(self, direccion):
        self._direccion = direccion

    def getEdad(self):
        return self._edad

    def setEdad(self, edad):
        self._edad  = edad

    def getHistorialPedidos(self):
        return self._historialPedidos

    def setHistorialPedidos(self, historial):
        self._historialPedidos = historial

    @classmethod
    def getClientes(cls):
        return cls._clientes

    @classmethod
    def setClientes(cls, clientes):
        cls._clientes = clientes

    # Metodos

    def agregarPedidoHistorial(self,  pedido):
        self._historialPedidos.append(pedido)
    
    # Implementacion de la interfaz

    def informacion(self):
        return f"El cliente {self._nombre} tiene email {self._email} y telefono {self._telefono}.\n" \
               f"Tiene {self._edad} anos.\n" \
               f"Vive en la direccion {self._direccion}.\n" \
               f"Ha hecho {len(self._historialPedidos)} pedidos en la aplicacion."

File: Practica1_i-Lunch/src/gestorAplicacion/usuariosRestaurante/Cliente.java
package gestorAplicacion.usuariosRestaurante;

import java.util.ArrayList;
import java.io.Serializable;

import gestorAplicacion.gestionRestaurante.*;

public class Cliente implements Serializable, Usuario {

	// Serializacion
	private static final long serialVersionUID = 1L;
	private static ArrayList<Cliente> clientes;
	static {
		clientes = new ArrayList<Cliente>();
	}

	// Atributos
	private int telefono;
	private String nombre;
	private String direccion;
	private int edad;
	private String correoElectronico;
	// ArrayList en el cual se agregaran todos los pedidos hechos por el cliente a
	// lo largo del tiempo.
	private ArrayList<Pedido> historialPedidos = new ArrayList<Pedido>();

	public Cliente(int telefono, String nombre, String direccion, int edad, Pedido pedidoActivo,
			ArrayList<Pedido> historialPedidos, String correo) {
		super();
		this.telefono = telefono;
		this.nombre = nombre;
		this.direccion = direccion;
		this.edad = edad;
		this.pedidoActivo = pedidoActivo;
		this.historialPedidos = historialPedidos;
		this.correoElectronico = correo;
		clientes.add(this);
	}

	public Cliente() {
		this(0, "", "", 0, null, null, null);
	}

	// Aca se almacenara el pedido que este la persona realizando en el momento,
	// al despachar o cancelar dicho pedido este atributo tiene que volver a null.
	// y en particular despues de ser despachado este pedido activo tiene que ser
	// agregado al historial.
	private Pedido pedidoActivo = null;

	///////// GETTERS AND SETTERS /////////
	public int getTelefono() {
		return telefono;
	}

	public void setTelefono(int telefono) {
		this.telefono = telefono;
	}

	public String getNombre() {
		return nombre;
	}

	public void setPrimerNombre(String nombre) {
		this.nombre = nombre;
	}

	public String getDireccion() {
		return direccion;
	}

	public void setDireccion(String direccion) {
		this.direccion = direccion;
	}

	public static ArrayList<Cliente> getClientes() {
		return clientes;
	}

	public Pedido getPedidoActivo() {
		return pedidoActivo;
	}

	public void setPedidoActivo(Pedido pedidoActivo) {
		this.pedidoActivo = pedidoActivo;
	}

	public ArrayList<Pedido> getHistorialPedidos() {
		return historialPedidos;
	}

	public void setHistorialPedidos(ArrayList<Pedido> historialPedidos) {
		this.historialPedidos = historialPedidos;
	}

	public int getEdad() {
		return edad;
	}

	public void setEdad(int edad) {
		this.edad = edad;
	}

	public void setNombre(String nombre) {
		this.nombre = nombre;
	}

	public String getCorreoElectronico() {
		return correoElectronico;
	}

	public void setCorreoElectronico(String correoElectronico) {
		this.correoElectronico = correoElectronico;
	}

	public static void setClientes(ArrayList<Cliente> clientes) {
		Cliente.clientes = clientes;
	}

	// Implementación de la interfaz Usuario
	public String informacion() {
		return "El cliente " + this.nombre + " con email " + this.correoElectronico + " y teléfono " + this.telefono
				+ " vive en la dirección " + this.direccion + " y tiene " + this.edad + " años.\n" + "Ha hecho "
				+ this.historialPedidos.size() + " pedidos en la aplicacion.";
	}

	public void agregarPedidoHistorial(Pedido pedido) {
		historialPedidos.add(pedido);
	}
}
